src/coord/src/coord.rs

```diff
@@ -2338,11 +2320,10 @@ where
                 // Continue to do those things.
                 for source_id in source_ids {
                     self.update_timestamper(source_id, true).await;
-                    let since_ts = since_ts.unwrap_or(0);
 
                     let frontiers = self.new_source_frontiers(
                         source_id,
-                        [since_ts],
+                        [0],
```

Comment:
This line looks superficially scary, in that if there is any reason at all that `persist` has an opinion on `since_ts` we are now ignoring it. There *are* reasons for STORAGE to want to set this to something other than zero, for example re-loading an already timestamped and compacted source, which for reasons of definiteness needs to start with `since_ts` at whatever it was compacted to.

Reply:
I agree that we should keep heeding the `since_ts` we get from persist, even though in practice it is always (_I think_) `0` right now, when sequencing the `CREATE SOURCE`. Just because there might already be data in persist, in the future, when we create the source.

Reply:
I *think* that what we want is for the catalog (eventually) to own recording the description of the source from a SQL point of view, whatever allows it to reconstruct `SourceDesc`, and then for STORAGE to own "thank you for creating this source; I will now reveal its properties to you" which would include initial setting of `since_ts`, maintenance of `upper`, and providing ongoing access to "here is a `SourceInstantiation` plan for how to build this" (this also allowing STORAGE to reveal how much of MFPs it wants to absorb).

I have another PR that has the beginnings of this, and it would be great to learn if the changes here are temporary varnish that we are going to rip up anyhow as soon as we figure out how to allow the STORAGE client to do the above things, and if that change is soon, we could skip the risk of regressions here. It could be that the change I'm thinking of is blocked on moving persist out of coord and in to STORAGE though, which is not imminent.

Reply:
Yeah, the point was previously this code was looking at a `persist: Option<SerializedPersistDetails>` that was _always_ `None`. Like, the code was attempting to ask persistence about the frontier, but persist hadn't even been given a chance to provide an opinion about the frontier, AFAICT.